src/webgpu/api/validation/encoding/programmable/pipeline_bind_group_compat.spec.ts

```diff
@@ -141,3 +213,129 @@ g.test('it_is_invalid_to_draw_in_a_render_pass_with_missing_bind_groups')
       commandEncoder.finish();
     }, !_success);
   });
+
+g.test('buffer_binding,render_pipeline')
+  .desc(
+    `
+  The GPUBufferBindingLayout bindings configure should be exactly
+  same in PipelineLayout and bindgroup.
+  - TODO: test more draw functions, e.g. indirect
+  - TODO: test more visibilities, e.g. vetex
+  - TODO: bind group should be created with different layout
+  `
+  )
+  .params(u => u.combine('type', kBufferBindingTypes))
+  .fn(async t => {
+    const { type } = t.params;
+
+    // Create fixed bindGroup
+    const uniformBuffer = t.getUniformBuffer();
+
+    const bindGroup = t.device.createBindGroup({
+      entries: [
+        {
+          binding: 0,
+          resource: {
+            buffer: uniformBuffer,
+          },
+        },
+      ],
+      layout: t.device.createBindGroupLayout({
+        entries: [
+          {
+            binding: 0,
+            visibility: GPUShaderStage.FRAGMENT,
+            buffer: {}, // default type: uniform
```

Comment:
nit: the comment could be better conveyed by just specifying the same thing in code. (I don't think the defaulting is needed for the test coverage)
```suggestion
            buffer: { type: 'uniform' },
```